Automattic/jetpack
# My Jetpack: Add ActivationScreen component to allow redeeming licenses

#### Changes proposed in this Pull Request:

Adds ActivationScreen Component to `add-license` route of My Jetpack so that we can functionally attach licenses

**Note**: This UI is still behind a feature flag in My Jetpack.

There are some minor differences that I'll point out from the licensing activation UI here in My Jetpack vs what's in the Jetpack plugin at the moment:

- The Jetpack logo in My Jetpack is 40px due to a global My Jetpack style. I chose not to override this. It is 48px in the Jetpack plugin.
- The "Contact Us" link in My Jetpack has an underline. In general, this seems to be the case in My Jetpack, so I chose not to override it.
- In My Jetpack, licensing activation currently has the back button and the footer elements. I think the back button is a good add, unclear about the footer. I was going for consistency within My Jetpack.

Something else that I'll point out is that the box shadow of this activation screen is different than for the rest of My Jetpack. We may want to update that, but I'd suggest doing it in a follow-up PR.


#### Jetpack product discussion



NA. This is necessary so that we can attach licenses from the standalone plugins as opposed to requiring the Jetpack plugin to attach a license.

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






To see the licensing activation flow in its original location, do the following:

- Go to the Jetpack Dashboard page
- Click "My Plan"
- Click blue "Activate a Product" button

Here's a screenshot of what that looks like:

<img width="1508" alt="Screen Shot 2022-04-22 at 9 37 56 AM" src="https://user-images.githubusercontent.com/1126811/164736713-c7be2abe-776b-4328-adad-6744aecf4056.png">


For testing the changes in this PR:

- Checkout PR
- `jetpack build packages/my-jetpack`
- Go to `$site.com/wp-admin/admin.php?page=my-jetpack#/add-license` and try to enter a license that will not validate... like `IAMBATMAN`
- Verify that you can get an error about an invalid license: `Jetpack: [invalid_input] Unable to validate license.`
- Go to https://cloud.jetpack.com/pricing and purchase a product (anything but CRM or Search)
- After purchasing, when selecting the site, scroll to the bottom for manual setup
- Copy the license code, enter into the input, then submit
- Verify that you get a success screen
- Click "back" button to go to My Jetpack page and verify that your new plan shows. Refresh the page and ensure the plan does not show immediately.

<img width="1141" alt="Screen Shot 2022-04-18 at 12 46 08 PM" src="https://user-images.githubusercontent.com/1126811/163850407-eff2e5fe-a226-493e-a70c-1fe79cd227bf.png">
<img width="1142" alt="Screen Shot 2022-04-18 at 12 45 57 PM" src="https://user-images.githubusercontent.com/1126811/163850411-b3a1aca5-e9b4-4a32-a437-5ce43e496725.png">
<img width="1145" alt="Screen Shot 2022-04-18 at 12 46 30 PM" src="https://user-images.githubusercontent.com/1126811/163850403-897939e8-95ca-4e8e-9127-813e8219a827.png">
